### ISSUE TITLE
Cannot query specific page (eg. in Queue)

### ISSUE THREAD

**Issue by Ezwen** [devopsarr/sonarr-py#41]
In many cases, the Sonarr API expects a `page` parameter when getting data that spans over multiple pages.

For example, this curl command is valid:

```
$ curl   -H "X-Api-Key: XXXXX" -X GET "http://my.local.sonarr:8989/api/v3/queue?page=2"
```

and successfully returns the second page of the queue.

However I have tried with sonarr-py:

```python
import sonarr
configuration = sonarr.Configuration(
    host = "http://my.local.sonarr:8989"
)
configuration.api_key['X-Api-Key'] = "XXXXX"
with sonarr.ApiClient(configuration) as api_client:
    sonarr.QueueApi(api_client).get_queue(page=2)
```

but this returns
```
sonarr.exceptions.ApiTypeError: Got an unexpected keyword argument 'page' to method get_queue
```

Maybe there is a way I did not understand?

**Comment by Fuochi** [comment]
Hi @Ezwen,
Unfortunately, sonarr doesn't specify the query parameter `page` in their swagger file:
https://github.com/Sonarr/Sonarr/blob/447d8cfbb9a7a9b5e3f2f985e7300c1d2b251388/src/Sonarr.Api.V3/openapi.json#L5214-L5239
and this library is based directly on that.
Right now there is no way to add a custom query parameter because they're initialized here: https://github.com/devopsarr/sonarr-py/blob/a3f53b60b74f3c4f576ba5936d3a353eef40768f/sonarr/api/queue_api.py#L473-L480

To make this work I'll have to put a workaround when generating the sdk to add pages in some cases.
I should be able to do that quite easily for:

- `/blocklist`
- `/wanted/cutoff`
- `/history`
- `/log`
- `/wanted/missing`
- `/queue`

 But it may take me some time to put this in place.
 
 Are there any other endpoints you think this would be needed?

**Comment by Ezwen** [comment]
>  Unfortunately, sonarr doesn't specify the query parameter page in their swagger file

Yes I noticed that and I suspected this was the problem here… strange from sonarr's side! 

I'll open an issue on their bug tracker to ask about that.

>  Are there any other endpoints you think this would be needed?

I checked in their openapi definition and I found the exact same list of endpoints as you :+1:

**Comment by Fuochi** [comment]
Hi @Ezwen,
they updated the swagger file, thanks for the ticket! this should be fixed in v0.10.0, could you please check? I'll wait for your feedback before releasing on other SDKs
